pkg/scheduler/plugins/nodeorder/nodeorder.go

```diff
@@ -302,23 +301,29 @@ func (pp *nodeOrderPlugin) OnSessionOpen(ssn *framework.Session) {
 	batchNodeOrderFn := func(task *api.TaskInfo, nodeInfo []*api.NodeInfo) (map[string]float64, error) {
 		// InterPodAffinity
 		state := k8sframework.NewCycleState()
+		nodeInfos := make([]*k8sframework.NodeInfo, 0, len(nodeInfo))
+		for _, node := range nodeInfo {
+			newNodeInfo := &k8sframework.NodeInfo{}
+			newNodeInfo.SetNode(node.Node)
+			nodeInfos = append(nodeInfos, newNodeInfo)
+		}
 		nodes := make([]*v1.Node, 0, len(nodeInfo))
```

Comment:
Is line305-line309 still needed? there are two for loop here, we should reduce the complexity.

Reply:
Thanks. I have move line305 -309  to the following loop body.